src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,433 @@
+		if (idDemandeAvenant != null) {
+			Optional<DemandeAvenantPivot> oDemandeAvenant = getDemandeAvenant(idDemandeAvenant);
+			if (oDemandeAvenant.isPresent()) {
+				DemandeAvenantPivot demandeAvenant = oDemandeAvenant.get();
+				Optional<DemandeAvenantContratEntity> oEntityContrat = repositoryDemandeAvenantContrat.findById(idDemandeAvenant);
+				if (oEntityContrat.isPresent()) {
+					demandeAvenant.setCodeProduit(oEntityContrat.get().getVersionProduit());
+
+					Optional<DemandeAvenantPivot> oContrat = convContrat.entityContratToContrat(oEntityContrat, oDemandeAvenant);
+					Optional<RolePersonnePivot> oAssure = cPersonne.getAssure(idDemandeAvenant);
+					Optional<RolePersonnePivot> oAdherent = cPersonne.getAdherent(idDemandeAvenant);
+					List<GarantiePivot> garanties = cGarantie.getGarantiesV3(idDemandeAvenant);
+
+					ContratPivot contratAlimente = null;
+					if (StringUtils.equals(demandeAvenant.getEtat().getCode(), EtatDemandeAvenant.PHOTO.getCode())) {
+						demandeAvenant.setPhotoDuContratALOuvertureDemande(oContrat.get().getDemandeAvenant());
+						contratAlimente = demandeAvenant.getPhotoDuContratALOuvertureDemande();
+					} else {
+						demandeAvenant.setDemandeAvenant(oContrat.get().getDemandeAvenant());
+						contratAlimente = demandeAvenant.getDemandeAvenant();
+					}
+					if (contratAlimente != null) {
+						contratAlimente.getInformations().setRoles(Arrays.asList(oAssure, oAdherent).stream().filter(Optional::isPresent).map(Optional::get).collect(Collectors.toList()));
+						contratAlimente.getInformations().setGaranties(garanties);
+					}
+				}
+				return oDemandeAvenant;
+			}
+		}
+		return Optional.empty();
+	}
+
+	public void fillContratEtGaranties(BigInteger idDemandeAvenant, Optional<DemandeAvenantPivot> oDemandeAvenant) {
+		DemandeAvenantPivot demandeAvenant = oDemandeAvenant.get();
+		Optional<DemandeAvenantContratEntity> oEntityContrat = repositoryDemandeAvenantContrat.findById(idDemandeAvenant);
+		if (oEntityContrat.isPresent()) {
+			demandeAvenant.setCodeProduit(oEntityContrat.get().getVersionProduit());
+
+			Optional<DemandeAvenantPivot> oContrat = convContrat.entityContratToContrat(oEntityContrat, oDemandeAvenant);
+			List<GarantiePivot> garanties = cGarantie.getGarantiesV3(idDemandeAvenant);
+			ContratPivot contratAlimente = null;
+			if (StringUtils.equals(demandeAvenant.getEtat().getCode(), EtatDemandeAvenant.PHOTO.getCode())) {
+				demandeAvenant.setPhotoDuContratALOuvertureDemande(oContrat.get().getDemandeAvenant());
+				contratAlimente = demandeAvenant.getPhotoDuContratALOuvertureDemande();
+			} else {
+				demandeAvenant.setDemandeAvenant(oContrat.get().getDemandeAvenant());
+				contratAlimente = demandeAvenant.getDemandeAvenant();
+			}
+			if (contratAlimente != null) {
+				contratAlimente.getInformations().setGaranties(garanties);
+			}
+		}
+	}
+
+	public Optional<IntervalleMontantsPivot> getIntervalleDeModificationDuRevenuAssure(BigDecimal revenuAssure) {
```

Comment:
The line is too long and should be split for better readability and maintainability. (Severity: low)